packages/ui/feature-builder-form-controls/src/lib/components/refreshable-property-core/refreshable-property-core-control.component.ts

```diff
@@ -0,0 +1,168 @@
+      return combineLatest(this.getPropertyRefreshers(true)).pipe(
+        startWith(this.parentFormGroup.value),
+        debounceTime(300)
+      );
+    }))
+    const search$ = this.getSearchRefresher();
+    const singleTimeRefresher$ = of('singleTimeRefresher');
+    return combineLatest({
+      refreshers: refreshers$,
+      search: search$.pipe(startWith('')),
+      singleTimeRefresher: singleTimeRefresher$,
+    }).pipe(
+      switchMap((res) => {
+        return this.piecetaDataService
+          .getPieceActionConfigOptions<T>({
+            flowId: this.flow.id,
+            flowVersionId: this.flow.version.id,
+            input: {
+              ...res.refreshers,
+            },
+            packageType: this.pieceMetaData.packageType,
+            pieceName: this.pieceMetaData.name,
+            pieceType: this.pieceMetaData.pieceType,
+            pieceVersion: this.pieceMetaData.version,
+            propertyName: this.propertyName,
+            stepName: this.actionOrTriggerName,
+            searchValue: res.search,
+          })
+          .pipe(
+            tap(() => {
+              this.loading$.next(false);
+            }),
+          );
+      }),
+      shareReplay(1),
+    );
+  }
+
+  private getSearchRefresher() {
+    return this.property.type === PropertyType.DROPDOWN &&
+      this.property.refreshOnSearch &&
+      this.searchRefresher$
+      ? this.searchRefresher$
+      : of('');
+  }
+
+  private getPropertyRefreshers(
+    startWithInitialValue: boolean,
+  ): Record<string, Observable<unknown>> {
+    const refreshers =
+      this.property.refreshers
+        .map((refresherName) => {
+          const control = this.parentFormGroup.get(refresherName);
+          const refresh$ = control?.valueChanges;
+          if (refresh$) {
+            return {
+              [refresherName]: startWithInitialValue
+                ? refresh$.pipe(startWith(control.value))
+                : refresh$,
+            };
+          } else {
+            console.error(
+              `Refreshable dropdown control: ${this.property.displayName} has a refresher ${refresherName} that does not exist in the form group`,
+            );
+            return null;
+          }
+        })
+        .filter((refresher$) => refresher$ !== null)
+        .reduce((acc, curr) => {
+          return { ...acc, ...curr };
+        }, {}) ?? {};
+    const authRefresher = this.parentFormGroup.get(
+      AUTHENTICATION_PROPERTY_NAME,
+    )?.valueChanges;
+    return {
+      ...refreshers,
+      ...spreadIfDefined(
+        AUTHENTICATION_PROPERTY_NAME,
+        startWithInitialValue
+          ? authRefresher?.pipe(
+              startWith(
+                this.parentFormGroup.get(AUTHENTICATION_PROPERTY_NAME)?.value,
+              ),
+            )
+          : authRefresher,
+      ),
+    };
+  }
+
+  protected refreshersChanged()
+  {
+    console.error("refreshersChanged not implemented")
```

Comment:
I dont unsee that, do you override it in children?

Reply:
yes.